vault/login_mfa.go

```diff
@@ -2029,23 +2035,65 @@ func (c *Core) validateOkta(ctx context.Context, mConfig *mfa.Config, username s
 		return fmt.Errorf("no MFA factors found for user")
 	}
 
-	var factorFound bool
+	var passcode string
+	if mfaFactors != nil {
+		passcode = mfaFactors.passcode
+	}
+
+	// When the user provides a passcode, we need to try each of their token* factors
+	// because we do not know which factor generated the token.
+	var passcodeUserFactorErrors map[*okta.UserFactor]error
+
 	var userFactor *okta.UserFactor
 	for _, factor := range factors {
 		if factor.IsUserFactorInstance() {
 			userFactor = factor.(*okta.UserFactor)
-			if userFactor.FactorType == "push" {
-				factorFound = true
-				break
+			switch userFactor.FactorType {
+			case "push":
+				if passcode != "" {
+					// push does not use passcode
+					continue
+				}
+
+				// FactorType "push" requires 2 steps to verify the factor:
+				// https://developer.okta.com/docs/reference/api/factors/#issue-a-push-factor-challenge
+				// https://developer.okta.com/docs/reference/api/factors/#verify-a-push-factor-challenge
+				return c.validateOktaPush(ctx, oktaClient, orgURL, user, userFactor)
+			case "token:software:totp", "token:hotp", "token", "token:hardware":
```

Comment:
Okta has a variety of factor types that accept a TOTP. Technically, there is no difference in what we have to do to support them, so I added all of them.
https://developer.okta.com/docs/reference/api/factors/#factors-that-require-only-a-verification-operation

This is particularly noteworthy as it probably implements #22961 since the Okta docs say that YUBICO uses `token:hardware`.

Maybe in the future we can also add `webauthn`/`u2f` support, but I did not add that here as it uses a slightly different API flow.